### ISSUE TITLE
Decompiling zip

### ISSUE THREAD

**Issue by me0wday** [MasterBugPatch/JDeobT#1]
How to use this to decompile a zip with pre-decompiled jad files? or java files?

Problem is, the JAR i am trying to deob has duplicate file names, and the extraction process does not work.

**Comment by Ruegg** [comment]
If the jar has duplicate names, you should first use a tool such as JMD which will rename all the classes in a jar file, https://github.com/contra/JMD .

Once you do that all you need to do is, open up your terminal(most likely command prompt), then 'cd' to the directory where JDeobT is. 
You would do something like 'cd C:/Users/thinkv/desktop/' if JDeobT is on your desktop. 

Once you have done that, you run 'java -jar 1.1.2.jar' (1.1.2) is the most recent jdeobt file name from ByteStructure.com . 

Lastly you would do 'C:/Directory/To/Target.jar -jar -zkm' . Use -zkm for Zelix Klassmaster and -allatori for Allatori.

If you have any problems feel free to tell me

**Comment by me0wday** [comment]
Unfortunately I have already tried JMD for the renamer and I am just
getting errors :(
Exception in thread "main" org.apache.bcel.classfile.ClassFormatException:
Invalid byte tag in constant pool: 15

**Comment by Ruegg** [comment]
Hm, well most OS can't have duplicate file names as such, so unless you can find a renamer for your classes I don't believe it will work

**Comment by me0wday** [comment]
Would you be able to modify to allow running on unix based system? it
screws up when looking for jad.exe.

**Comment by Ruegg** [comment]
Does it throw any errors?

**Comment by me0wday** [comment]
Get's the directory all wrong, and seems to make some weird sub directory,
then doesn't seem to do anything.

**Comment by mrdrako** [comment]
I think I've got the same error going on. It creates a whole new directory structure for my path, with a bin folder at the end. Then when it processes the commands, it fails with the following:

Beggining deobfuscation
Exception in thread "main" java.lang.IllegalArgumentException: Parameter 'directory' is not a directory
    at org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:545)
    at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:521)
    at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:691)
    at bugpatch.master.JDeobT.commandExecuted(JDeobT.java:162)
    at bugpatch.master.JDeobT.getInput(JDeobT.java:34)
    at bugpatch.master.JDeobT.main(JDeobT.java:24)

The file I am trying to look at is an allatori obfuscated malware jar file, and uses case-sensitive file names. This fails on both Windows 7 and on Ubuntu 15.04, both using java 8u45. I also tried the JMD renaming trick, same issue (and it shouldn't matter on Ubuntu as it is case-sensitive).

I can provide the malware sample to you securely if you want it for testing.
